src/interaction/command/circle.ts

```diff
@@ -0,0 +1,260 @@
+import Bot from "../../api/bot";
+import {
+  CategoryChannel,
+  ColorResolvable,
+  CommandInteraction,
+  GuildMember,
+  OverwriteResolvable,
+} from "discord.js";
+import { settings } from "../../settings";
+import { CircleData } from "../../api/schema";
+import SlashCommand, {
+  SlashCommandContext,
+} from "../../api/interaction/slashcommand";
+
+export default class CircleCommand extends SlashCommand {
+  public constructor() {
+    super({
+      name: "circle",
+      description: "A suite of command that manage ACM Community Circles.",
+      permissions: [
+        { id: settings.roles.director, permission: true, type: "ROLE" },
+      ],
+    });
+    // Adding "add" subcommand
+    this.slashCommand.addSubcommand((subcommand) => {
+      subcommand.setName("add");
+      subcommand.setDescription("Add a new circle.");
+      // Set up all the options for the subcommand
+      // Name, description, color, emoji, graphic, and owner
+      subcommand.addStringOption((option) => {
+        option.setName("name");
+        option.setDescription("The name of the circle.");
+        option.setRequired(true);
+        return option;
+      });
+      subcommand.addStringOption((option) => {
+        option.setName("description");
+        option.setDescription("The description of the circle.");
+        option.setRequired(true);
+        return option;
+      });
+      subcommand.addStringOption((option) => {
+        option.setName("color");
+        option.setDescription(
+          "The color of the circle (used for role and embeds)"
+        );
+        option.setRequired(true);
+
+        option.addChoices(colorOptions as [string, string][]);
```

Comment:
we generally want the colors to be somewhat distinct, so have been using hex codes for the colors